cmd/noms/noms.go

```diff
@@ -44,49 +29,70 @@ var actions = []string{
 	"nomming on",
 }
 
-func usageString() string {
-	i := rand.New(rand.NewSource(time.Now().UnixNano())).Intn(len(actions))
-	return fmt.Sprintf(`Noms is a tool for %s Noms data.`, actions[i])
+// CommandHandler - a callback passed to commands
+type CommandHandler func() (exitCode int)
+
+var (
+	verboseFlag *bool
+	quietFlag   *bool
+)
+
+// AddVerboseFlags adds --verbose and --quiet flags to the passed command
+func AddVerboseFlags(cmd *kingpin.CmdClause) (verboseFlag *bool, quietFlag *bool) {
+	verboseFlag = cmd.Flag("verbose", "show more").Short('v').Bool()
+	quietFlag = cmd.Flag("quiet", "show less").Short('q').Bool()
+	return
+}
+
+// ApplyVerbosity - run when commands are invoked to apply the verbosity arguments configured in AddVerboseFlags
+func ApplyVerbosity() {
+	verbose.SetVerbose(*verboseFlag)
+	verbose.SetQuiet(*quietFlag)
+}
+
+// AddDatabaseArg adds a "database" arg to the passed command
+func AddDatabaseArg(cmd *kingpin.CmdClause) (arg *string) {
+	return cmd.Arg("database", "a noms database path").Required().String() // TODO: custom parser for noms db URL?
+}
+
+// Commands, in order of preference
+var commands = []func(*kingpin.Application) (*kingpin.CmdClause, CommandHandler){
+	NomsCommit,
+	NomsConfig,
+	NomsDiff,
+	NomsDs,
+	NomsLog,
+	NomsMerge,
+	NomsRoot,
+	NomsServe,
+	NomsShow,
+	NomsSync,
+	NomsVersion,
 }
 
 func main() {
-	util.InitHelp(path.Base(os.Args[0]), commands, usageString())
+	// allow short (-h) help
+	kingpin.CommandLine.HelpFlag.Short('h')
 
-	flag.Usage = util.Usage
-	flag.Parse(false)
+	// TODO: is there a way to dynamically generate help text other than re-initializing this every time?
+	i := rand.New(rand.NewSource(time.Now().UnixNano())).Intn(len(actions))
+	noms := kingpin.New("noms", fmt.Sprintf(`Noms is a tool for %s Noms data.`, actions[i]))
 
-	args := flag.Args()
-	if len(args) < 1 {
-		util.Usage()
-		return
-	}
+	handlers := make(map[string]CommandHandler)
```

Comment:
It's more idiomatic to say `map[string]CommandHandler{}`. Basically it's more idiomatic to use literals when possible, rather than `make`. `make` must be used for certain uncommon use cases (e.g., allocating a slice with a particular capacity to start), and thus, when a reader sees `make`, it's a flag that something interesting is about to happen. But nothing interesting is happening here :).